discovery_engine_core/core/src/engine.rs

```diff
@@ -444,7 +443,7 @@ impl Engine {
             stacks.values_mut().map(|s| s as _),
             once(&mut self.exploration_stack as _),
         )
-        .collect::<Vec<&mut dyn Bucket<_>>>();
+        .collect::<Vec<&mut (dyn Bucket<_> + Send)>>();
```

Comment:
unfortunately dropping it doesn't help, the error is still
```
generator cannot be sent between threads safely
the trait `Send` is not implemented for `dyn xayn_discovery_engine_core::mab::Bucket<xayn_discovery_engine_core::document::Document>`
```
also moving it into the selection iterator instead of just ref-muting it is not enough, but i refactored that a little bit such that it is at least a bit more ergonomic to use.